src/graphics/image.rs

```diff
@@ -204,10 +228,29 @@ impl Image {
     /// coordinates, using out-of-range values will result in
     /// an undefined behaviour.
     #[must_use]
-    pub unsafe fn pixel_at(&self, x: u32, y: u32) -> Color {
+    pub unsafe fn pixel_at_unchecked(&self, x: u32, y: u32) -> Color {
         ffi::sfImage_getPixel(self.image, x, y)
     }
 
+    /// Get the color of a pixel in an image
+    ///
+    /// # Arguments
+    /// * x - X coordinate of pixel to get
+    /// * y - Y coordinate of pixel to get
+    ///
+    /// Return the Color of the pixel at coordinates (x, y)
+    #[must_use]
+    pub fn pixel_at(&self, x: u32, y: u32) -> Option<Color> {
+        let image_size = self.size();
+        if image_size.x <= x || image_size.y <= y {
+            return None;
+        }
+
+        // Since we check for index validity before setting the pixel, it is safe unless the
```

Comment:
You mention setting the pixel here, but we're getting it. Probably copy-paste error.

Reply:
Fixed!